### PR TITLE
Port tests to CliTestContext

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

File: tests/suite/cli_exact.rs [hunk 1]
```diff
@@ -3,29 +3,24 @@
...
 #[test]
 fn update_once() {
-    test(&|config| {
-        config.with_scenario(Scenario::SimpleV2, &|config| {
-            config.expect_ok_ex(
-                &["rustup", "update", "nightly"],
-                for_host!(
-                    r"
+    let mut cx = CliTestContext::from(Scenario::None);
+    cx.config.with_scenario(Scenario::SimpleV2, &|config| {
```
Comment (rami3l) [inline tests/suite/cli_exact.rs]: Just being curious, how do you plan to further simplify this case?

**Review by rami3l — APPROVED** [review]
LGTM :)

File: tests/suite/cli_exact.rs (hunk 1 above)
Comment (djc) [reply tests/suite/cli_exact.rs]: I think for most of these we can just pass `Scenario::SimpleV2` as the initial scenario... Trying now to remove `with_scenario()` after I made an initial pass porting all of these over to using `CliTestContext`.

Comment (rami3l) [reply tests/suite/cli_exact.rs]: @djc I've rebased the PR to address formatting errors, and ~~I'm working on migrating~~ I have migrated `cli_v2` as well.

File: src/cli/self_update/windows.rs [hunk 2]
```diff
@@ -807,16 +808,85 @@ pub(crate) fn delete_rustup_and_cargo_home(process: &Process) -> Result<()> {
...
+            Err(ref e) if e.kind() == io::ErrorKind::NotFound => Ok(None),
+            Err(e) => Err(e),
+        }
+    }
+
+    pub fn set_value(&self, new: Option<impl ToRegValue>) -> io::Result<()> {
+        self.set(new.map(|s| s.to_reg_value()).as_ref())
+    }
+
+    pub fn set(&self, new: Option<&RegValue>) -> io::Result<()> {
```
Comment (rami3l) [inline src/cli/self_update/windows.rs]: Question: Do we really need to put both `.set()` and `.set_value()` in the public API?

**Review by rami3l — APPROVED** [review]
Thanks again for the tremendous amount of work (~14 000 diffLoC)! Wouldn't be possible without you!

File: src/cli/self_update/windows.rs (hunk 2 above)
Comment (djc) [reply src/cli/self_update/windows.rs]: We do not -- good point! Made `get()` and `set()` private.

**Comment by djc** [comment]
Too bad this is like +1000 lines... all from adding async in the CLI tests. Oh well...

**Pull request merged**